src/big/big_rational.cr

```diff
@@ -360,6 +360,13 @@ struct BigRational < Number
     to_s io
   end
 
+  # :inherit:
+  def format(io : IO, separator = '.', delimiter = ',', decimal_places : Int? = nil, *, group : Int = 3, only_significant : Bool = false) : Nil
+    io << self.denominator.format(separator, delimiter, decimal_places, group: group, only_significant: only_significant)
+    io << '/'
+    io << self.numerator.format(separator, delimiter, decimal_places, group: group, only_significant: only_significant)
```

Comment:
**issue:** This still allocates intermediate format strings. The idea is to format directly to `io` by passing it to `format`:
```suggestion
    self.denominator.format(io, separator, delimiter, decimal_places, group: group, only_significant: only_significant)
    io << '/'
    self.numerator.format(io, separator, delimiter, decimal_places, group: group, only_significant: only_significant)
```